File: app/src/test/java/org/schulcloud/mobile/data/model/jsonApi/IncludedAttributesTest.java
package org.schulcloud.mobile.data.model.jsonApi;

import org.junit.Before;
import org.junit.Test;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertNotNull;

public class IncludedAttributesTest {
    private static final String FREQ = "FREQ";
    private static final String UNTIL = "UNTIL";
    private static final String WKST = "WKST";

    private IncludedAttributes includedAttributes;

    @Before
    public void setUp() throws Exception {
        includedAttributes = newIncludedAttributes();
    }

    public static IncludedAttributes newIncludedAttributes() {
        IncludedAttributes includedAttributes = new IncludedAttributes();
        includedAttributes.setFreq(FREQ);
        includedAttributes.setUntil(UNTIL);
        includedAttributes.setWkst(WKST);

        return includedAttributes;
    }

    @Test
    public void getFreq() throws Exception {
        assertEquals(includedAttributes.getFreq(), FREQ);
    }

    @Test
    public void setFreq() throws Exception {
        includedAttributes.setFreq(FREQ);
        assertNotNull(includedAttributes.getFreq());
    }

    @Test
    public void getUntil() throws Exception {
        assertEquals(includedAttributes.getUntil(), UNTIL);
    }

    @Test
    public void setUntil() throws Exception {
        includedAttributes.setUntil(UNTIL);
        assertNotNull(includedAttributes.getUntil());
    }

    @Test
    public void getWkst() throws Exception {
        assertEquals(includedAttributes.getWkst(), WKST);
    }

    @Test
    public void setWkst() throws Exception {
        includedAttributes.setWkst(WKST);
        assertNotNull(includedAttributes.getWkst());
    }

}

File: app/src/test/java/org/schulcloud/mobile/models/event/IncludedAttributesTest.kt
package org.schulcloud.mobile.models.event

import org.junit.Before

import org.junit.Assert.*
import org.junit.Test

class IncludedAttributesTest {
    private companion object {
        const val FREQ = "freq"
        const val UNTIL = "until"
        const val WEEKDAY = "weekday"
        const val INVALID_WEEKDAY = "invWeekday"
        const val VALID_WEEKDAY = "SU"
    }

    private val newIncludedAttributes: IncludedAttributes
        get() = IncludedAttributes().apply {
            freq = FREQ
            until = UNTIL
            weekday = WEEKDAY
        }
    private lateinit var includedAttributes: IncludedAttributes

    @Before
    fun setUp() {
        includedAttributes = newIncludedAttributes
    }

    @Test
    fun testGetProperties() {
        assertEquals(FREQ, includedAttributes.freq)
        assertEquals(UNTIL, includedAttributes.until)
        assertEquals(WEEKDAY, includedAttributes.weekday)
    }

    @Test
    fun testWeekdayNumberNotNullWhenWeekdayValid() {
        includedAttributes.weekday = VALID_WEEKDAY
        assertNotNull(includedAttributes.weekdayNumber)
    }

    @Test
    fun testWeekdayNumberNullWhenWeekdayInvalid() {
        includedAttributes.weekday = INVALID_WEEKDAY
        assertNull(includedAttributes.weekdayNumber)
    }
}
